source/common/protobuf/utility.h

```diff
@@ -48,10 +49,26 @@ class RepeatedPtrUtil {
 class MessageUtil {
 public:
   static std::size_t hash(const Protobuf::Message& message) {
-    return std::hash<std::string>{}(message.SerializeAsString());
+    std::string text;
+    Protobuf::io::StringOutputStream string_stream(&text);
+    Protobuf::io::CodedOutputStream coded_stream(&string_stream);
+    coded_stream.SetSerializationDeterministic(true);
+    message.SerializeToCodedStream(&coded_stream);
+    return std::hash<std::string>{}(text);
   }
 
+  static void loadFromJson(const std::string json, Protobuf::Message& message);
```

Comment:
Is the reason we are passing std::string here vs reference string issues again? If so can we put a TODO somewhere?